src/Microsoft.AspNet.Mvc.DataAnnotations/DataTypeClientModelValidatorProvider.cs

```diff
@@ -0,0 +1,33 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+
+namespace Microsoft.AspNet.Mvc.ModelBinding.Validation
+{
+    /// <summary>
+    /// An implementation of <see cref="IClientModelValidatorProvider"/> which provides client validators
+    /// for numeric types for which we set type='text'.
+    /// </summary>
+    public class DataTypeClientModelValidatorProvider : IClientModelValidatorProvider
```

Comment:
Maybe change this to `NumericDataTypeClientModelValidatorProvider` if this is specifically going to work with numeric types?